Assignment 2.Rmd

```diff
@@ -75,37 +82,93 @@ g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+V2.TCH <- EDGE%>%select(from,count)%>%group_by(from)%>%summarize(sum(count))
+names(V2.TCH) <- c("id","count")
+V2.STD <- EDGE%>%select(to,count)%>%group_by(to)%>%summarize(sum(count))
+names(V2.STD) <- c("id","count")
+B <- bind_rows(V2.TCH,V2.STD)
+VERTEX <- full_join(VERTEX, B, by="id")
+g2 <- graph.data.frame(EDGE, vertices=VERTEX, directed=TRUE)
+plot(g2, layout=layout.fruchterman.reingold, edge.arrow.size=0.28,edge.width=EDGE$count, vertex.color=VERTEX$gender, vertex.size=VERTEX$count, vertex.shapes=as.factor(VERTEX$group))
 
 ```
 
+## Part III
 
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 
-## Part III
 
-Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+
+
+h1 <- read.csv("hudk4050-classes.csv", header=TRUE)
+h1 <- lapply(h1, gsub, pattern=" ", replacement="")
+h1 <- as.data.frame(h1)
+h1 <- unite(h1, Name, "First.Name", "Last.Name", sep=" ")
+h1 <- filter(h1, Name > 0)
+h2 <- gather(h1, class_number, class_code, 2:7)
+h2 <- filter(h2, class_code > 0)%>%select(-class_number)
+h2 <- filter(h2, class_code != "HUDK4050")
+h2$count <- 1
+h2 <- unique(h2)
+h3 <- spread(h2, class_code, count, fill=0)
+row.names(h3) <- h3$Name
+h3$Name <- NULL
+h3 <- as.matrix(h3)
+h4 <- h3 %*% t(h3)
+diag(h4) <- NA
+g_PC <- graph.adjacency(h4, mode="undirected")
+plot(g_PC, layout=layout.fruchterman.reingold, vertex.label.size=0.428)
+
+```
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+degree(g_PC)
+betweenness(g_PC)
+d_N <- gather(h1, class_number, class, 2:7, na.rm=TRUE)%>% select(-class_number)
+d_N <- filter(d_N, class != "HUDK4050", class > 0)
+names(d_N) <- c("from","to")
+d_N <- unique(d_N)
+d_N$count <- 1
+
+v.std <- select(d_N, from, count)
+v.std <- group_by(v.std, from)%>%summarise(sum(count))
+v.std <- unique(v.std)
+v.cls <- select(d_N, to, count)
+v.cls <- group_by(v.cls, to)%>%summarise(sum(count))
+v.cls <- unique(v.cls)
+v.std$group <- "student"
+names(v.std) <- c("id","count","group")
+v.cls$group <- "class"
+names(v.cls) <- c("id", "count", "group")
+
+vertex_N <- bind_rows(v.std, v.cls)
+g_N<-graph.data.frame(d_N, directed = TRUE, vertices = vertex_N)
+plot(g_N, layout=layout.fruchterman.reingold,edge.arrow.size = 0.28, vertex.color=as.factor(vertex_N$group), vertex.size=vertex_N$count)
+
+```
+
+
```

Comment:
The effort you visualized the centrality is impressive. For this task, the function degree and betweenness would be sufficient.

Reply:
Great job! Keep up the good work.